lib/heroics/schema.rb

```diff
@@ -334,7 +334,39 @@ def lookup_parameter(path, schema)
     # @return [String] The formatted parameter.
     def format_parameter(parameter)
       formatted_parameter = parameter.instance_of?(Time) ? iso_format(parameter) : parameter.to_s
-      WEBrick::HTTPUtils.escape formatted_parameter
+      url_escape_parameter formatted_parameter
+    end
+
+    # URL escape a path parameter. Copied from WEBrick::HTTPUtils.escape:
+    # https://github.com/ruby/webrick/blob/master/lib/webrick/httputils.rb#L498
+    #
+    # @param [Fixnum,String,TrueClass,FalseClass,Time] The parameter to escape.
+    # @return [String] The escaped parameter.
+    def url_escape_parameter(parameter)
+      # Note: this RegEx is obtained by executing WEBrick::HTTPUtils::UNESCAPED,
+      # splitting it over multiple lines and using the x modifier to make this possible.
```

Comment:
I wonder if it would be better to either copy over the smaller pieces of this Regex, as WEBrick does it, or at least make that a perma-link to the version from which it was taken (i.e., `https://github.com/ruby/webrick/blob/9350944141a3f15acda9c79edd5393289c098e04/lib/webrick/httputils.rb#L498`). Or… both!?!

Reply:
I did not copy the smaller pieces of the regex but used the complete result instead on purpose. We do not need the smaller piecers. We do not use them by ourselves. It may improve understandability of where this is coming from, but to use the smaller piecer, we would also need to copy the custom `_make_regex` method and logic of what the UNESCAPED constant is (`UNESCAPED = _make_regex(control+space+delims+unwise+nonascii)`). This would result in duplicating more code.

Instead I think it is better to just execute WEBrick::HTTPUtils::UNESCAPED and copying the result as I did here.

Good suggestion about the permalink though. Fixed that :)